ChattoAdditions/Source/Chat Items/CompoundMessage/CompoundMessagePresenter.swift

```diff
@@ -0,0 +1,136 @@
+    private lazy var layoutProvider: CompoundBubbleLayoutProvider = self.makeLayoutProvider()
+    private let cache: Cache<CompoundBubbleLayoutProvider.Configuration, CompoundBubbleLayoutProvider>
+
+    public init(
+        messageModel: ModelT,
+        viewModelBuilder: ViewModelBuilderT,
+        interactionHandler: InteractionHandlerT?,
+        contentFactories: [AnyMessageContentFactory<ModelT>],
+        sizingCell: CompoundMessageCollectionViewCell<ModelT>,
+        baseCellStyle: BaseMessageCollectionViewCellStyleProtocol,
+        compoundCellStyle: CompoundBubbleViewStyleProtocol,
+        cache: Cache<CompoundBubbleLayoutProvider.Configuration, CompoundBubbleLayoutProvider>
+    ) {
+        self.compoundCellStyle = compoundCellStyle
+        self.contentFactories = contentFactories.filter { $0.canCreateMessageModule(forModel: messageModel) }
+        self.cache = cache
+        super.init(
+            messageModel: messageModel,
+            viewModelBuilder: viewModelBuilder,
+            interactionHandler: interactionHandler,
+            sizingCell: sizingCell,
+            cellStyle: baseCellStyle
+        )
+    }
+
+    public override var canCalculateHeightInBackground: Bool {
+        return true
+    }
+
+    public override class func registerCells(_ collectionView: UICollectionView) {
+        collectionView.register(CompoundMessageCollectionViewCell<ModelT>.self,
+                                forCellWithReuseIdentifier: .compoundCellReuseId)
+    }
+
+    public override func dequeueCell(collectionView: UICollectionView, indexPath: IndexPath) -> UICollectionViewCell {
+        return collectionView.dequeueReusableCell(withReuseIdentifier: .compoundCellReuseId,
+                                                  for: indexPath)
+    }
+
+    public override func heightForCell(maximumWidth width: CGFloat,
+                                       decorationAttributes: ChatItemDecorationAttributesProtocol?) -> CGFloat {
+        let layoutConstants = self.cellStyle.layoutConstants(viewModel: self.messageViewModel)
+        let maxWidth = (width * layoutConstants.maxContainerWidthPercentageForBubbleView)
+        return self.layoutProvider.makeLayout(forMaxWidth: maxWidth).size.height
+    }
+
+    public override func configureCell(_ cell: BaseMessageCollectionViewCell<CompoundBubbleView>,
+                                       decorationAttributes: ChatItemDecorationAttributes,
+                                       animated: Bool,
+                                       additionalConfiguration: (() -> Void)?) {
+        guard let compoundCell = cell as? CompoundMessageCollectionViewCell<ModelT> else {
+            assertionFailure("\(cell) is not CompoundMessageCollectionViewCell<\(ModelT.self)>")
+            return
+        }
+
+        super.configureCell(compoundCell, decorationAttributes: decorationAttributes, animated: animated) { [weak self] in
+            guard let self = self else { return }
```

Comment:
`sSelf` according to current code style

Reply:
Should we update the code style, since this is possible since Swift 4.2?

Reply:
We need to discuss it